### DISCUSSION TITLE
Does pp_utils work for mfusg model?

### DISCUSSION THREAD

**Discussion by hjia1005** [pypest/pyemu#420] [Q&A]
Does pp_utils work for mfusg model?

**Comment by hjia1005** [comment]
Based on this it should work. I will give it a go.
![image](https://user-images.githubusercontent.com/33449394/232395424-db56b341-a97c-476a-9364-b95b999478dd.png)

**Comment by jtwhite79** [comment]
Be careful here.  The unstructured grid support in pyemu is in general pretty weak...

**Reply by hjia1005** [reply]
Thank you, I will be careful~

**Reply by hjia1005** [reply]
Just report back here. I tried it out, had challenges with the function ok.calc_factors_grid( ), which needs the spatial_reference object to have the attributes xcentergrid and ycentergrid but it doesn't. The function calc_factors( ) can create the factors but has to assume only one zone for the whole model because NODEARRAY is not included in this function. May need to customize some codes to get it done. 

In the meantime, I used plproc in PEST utilities. I noticed that the kriging interpolation is much faster. I am not sure if it's because I used calc_kriging_factors_auto_2d( ) which speeds up the process?

**Reply by mnfienen** [reply]
hey @hjia1005  - glad you got things working. I'm not at all surprised that `plproc` is faster for the kiriging interpolation. At the moment, the kriging interpolation in `pyemu` is in python and much slower than the compiled `plproc`. It's something we are working on, but for now, the speedup will be greater with `plproc`.
